File: DownloadFiles/TGNet/Chapter08/Chapter8DemoCode/VB/ADODataReaderDemo_SQLServer/ADODataReaderDemo/Form1.vb
Imports System.Data.SqlClient

Public Class Form1
    'this code presumes SQL Server is installed locally and started and that the Northwind
    'database is available in the installation
    'This code logs into the SQL server with Window authentication
    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        Dim cn As New SqlConnection("Data Source=localhost;Database=Northwind;Trusted_Connection=Yes")
        Dim dr As SqlDataReader
        Dim cmd As New SqlCommand("Select * from customers where ContactName like 'M%' ", cn)

        cn.Open()
        dr = cmd.ExecuteReader
        If dr.HasRows Then
            Do While dr.Read
                Debug.WriteLine(dr.Item("ContactName"))
            Loop
            MessageBox.Show("To see answer: open the Immediate window!")
        Else
            Debug.WriteLine("no rows returned")
        End If
        dr.Close()
        cn.Close()

    End Sub
End Class

File: DownloadFiles/TGNet/Chapter08/Chapter8DemoCode/C#/ADODataReaderDemoSQLServer/ADODataReaderDemo/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.Data.SqlClient;
using System.Diagnostics;

namespace ADODataReaderDemo
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            SqlConnection cn = new SqlConnection
   ("Data Source=localhost;Database=Northwind;Trusted_Connection=Yes");
            SqlDataReader dr;
            SqlCommand cmd = new SqlCommand
               ("Select * from customers where ContactName like 'M%' ", cn);
            cn.Open();
            dr = cmd.ExecuteReader();
            if (dr.HasRows)
                while (dr.Read()==true)
                {
                    Debug.WriteLine(dr.GetString(dr.GetOrdinal("ContactName")));
                }
            else
                MessageBox.Show("no rows returned");
            dr.Close();
            cn.Close();

        }
    }
}